ingestion/src/metadata/cmd.py

```diff
@@ -331,127 +49,367 @@ def backup(
+        required=False,
+    )
+
+    parser.add_argument(
+        "--input",
+        help="Local backup file path for restore",
+        type=pathlib.Path,
+        default=None,
+    )
+
+    parser.add_argument("-o", "--options", default=None, action="append")
+
+    parser.add_argument("-a", "--arguments", default=None, action="append")
+
+    parser.add_argument(
+        "-s",
+        "--schema",
+        default=None,
+        required=False,
+    )
+
+
+def add_metadata_args(parser: argparse.ArgumentParser):
+    """
+    Addtional Parser Arguments for Metadata
+    """
+    parser.add_argument(
+        "-v", "--version", action="version", version=get_metadata_version()
+    )
+
+    parser.add_argument(
+        "-l",
+        "--log-level",
+        choices=["INFO", "DEBUG", "WARNING", "ERROR", "CRITICAL"],
+        help="Set Log Level",
+    )
+
+
+def metadata():
+    """
+    This method implements parsing of the arguments passed from CLI
     """
-    run_restore(
-        host,
-        user,
-        password,
-        database,
-        port,
-        sql_file,
-        options,
-        arguments,
-        schema,
+    parser = argparse.ArgumentParser(prog="metadata", description="Ingestion Framework")
+    sub_parser = parser.add_subparsers(dest="command")
+
+    create_common_config_parser_args(
+        sub_parser.add_parser("ingest", help="Ingestion Workflow", prog="ingest")
+    )
+    create_common_config_parser_args(
+        sub_parser.add_parser(
+            "profile", help="Workflow for profiling Table sources into Metadata"
+        )
+    )
+    create_common_config_parser_args(
+        sub_parser.add_parser("test", help="Workflow for running test suites")
     )
 
+    create_openmetadata_imports_migration_args(
+        sub_parser.add_parser(
+            "openmetadata_imports_migration",
+            help=OM_IMPORTS_MIGRATION,
+        )
+    )
+    docker_args(sub_parser.add_parser("docker", help="Docker Quickstart"))
+    backup_args(
+        sub_parser.add_parser(
+            "backup",
+            help=BACKUP_HELP,
+        )
+    )
+    restore_args(
+        sub_parser.add_parser(
+            "restore",
+            help=RESTORE_HELP,
+        )
+    )
+    webhook_args(
+        sub_parser.add_parser(
+            "webhook", help="Simple Webserver to test webhook metadata events"
+        )
+    )
+
+    add_metadata_args(parser)
+    parser.add_argument("--debug", help="Debug Mode", action="store_true")
+
+    contains_args = vars(parser.parse_args())
+    metadata_workflow = contains_args.get("command")
+    config_file = contains_args.get("config")
+    if contains_args.get("debug"):
+        set_loggers_level(logging.DEBUG)
+    elif contains_args.get("log_level"):
+        set_loggers_level(contains_args.get("log_level"))
+    else:
+        set_loggers_level(logging.INFO)
 
-metadata.add_command(check)
+    if metadata_workflow == "ingest":
+        run_ingest(config_path=config_file)
```

Comment:
could we maybe change this `if - else` chain into:

```python
if metadata_workflow == "ingest":
    run_ingest(config_path=config_file)
    return None
if ...
```

As there is no need to check the other branches once we get inside an if

Reply:
also the commands we can maybe create an `enum` and run the equal against `Commands.ingest.value` so we have a single place to control the allowed params